include/seqan/index/find2_index_approx.h

```diff
@@ -277,15 +302,26 @@ inline void _optimalSearchSchemeInit(std::array<OptimalSearch<nbrBlocks>, N> & s
     // for that we need to slightly modify search()
     for (OptimalSearch<nbrBlocks> & s : ss)
     {
-        s.startPos = 0;
-        for (uint8_t i = 0; i < s.pi.size(); ++i)
-            if (s.pi[i] < s.pi[0])
-                s.startPos += s.blocklength[i] - ((i > 0) ? s.blocklength[i-1] : 0);
+        if (s.pi[1] > s.pi[0])
+        {
+            s.startPos = 0;
+            for (uint8_t i = 1; i < s.pi.size(); ++i)
+                if (s.pi[i] < s.pi[0])
+                    s.startPos += s.blocklength[i] - s.blocklength[i-1];
+        }
+        else
```

Comment:
Is this block a fix, an extension or a brings-more-performance change?

Reply:
It should improve the performance a bit.